sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodegenExpressionCachingSuite.scala

```diff
@@ -109,3 +139,17 @@ case class MutableExpression() extends LeafExpression with CodegenFallback {
   override def nullable: Boolean = false
   override def dataType: DataType = BooleanType
 }
+
+/**
+ * An expression with evaluated state so we can know whether it is evaluated.
+ */
+case class MutableExpression2() extends LeafExpression with CodegenFallback {
```

Comment:
It seems that we need to give a more meaningful name instead of `MutableExpression2`.

Reply:
I'm not sure, but possibly `ExpressionWithEvaluatedFlag` or something?

Reply:
OK.